### ISSUE TITLE
tensor维度不匹配 Chapter3

### ISSUE THREAD

**Issue by Sonnet1219** [datawhalechina/hello-agents#61]
### 1. 遇到问题的章节 / Affected Chapter

Chapter3

### 2. 问题类型 / Issue Type

内容错误 / Content Error

### 3. 具体问题描述 / Problem Description

在transformer实现中 前面的MultiHeadAttention PositionWiseFeedForward的都是用新版(bsz, seq, dim)的x
例如：
```
class PositionWiseFeedForward(nn.Module):

    # 位置前馈网络模块

    def __init__(self, d_model, d_ff, dropout=0.1):
        super(PositionWiseFeedForward, self).__init__()
        self.linear1 = nn.Linear(d_model, d_ff)
        self.dropout = nn.Dropout(dropout)
        self.linear2 = nn.Linear(d_ff, d_model)
        self.relu = nn.ReLU()

    def forward(self, x):
        # x 形状: (batch_size, seq_len, d_model)
        x = self.linear1(x)
        x = self.relu(x)
        x = self.dropout(x)
        x = self.linear2(x)
        # 最终输出形状: (batch_size, seq_len, d_model)
        return x
```
但是在class PositionalEncoding(nn.Module)模块中构造的
pe = torch.zeros(max_len, 1, d_model)却沿用了RNN的(seq_len, bsz, dim)格式 希望统一一下维度 我认为应该是
pe = torch.zeros(1, max_len, d_model)

### 4. 问题重现材料 / Reproduction Materials

原文代码块：
```
class PositionalEncoding(nn.Module):

    # 为输入序列的词嵌入向量添加位置编码。

    def __init__(self, d_model: int, dropout: float = 0.1, max_len: int = 5000):
        super().__init__()
        self.dropout = nn.Dropout(p=dropout)

        # 创建一个足够长的位置编码矩阵
        position = torch.arange(max_len).unsqueeze(1)
        div_term = torch.exp(torch.arange(0, d_model, 2) * (-math.log(10000.0) / d_model))

        # pe (positional encoding) 的大小为 (max_len, d_model)
        pe = torch.zeros(max_len, 1, d_model)

        # 偶数维度使用 sin, 奇数维度使用 cos
        pe[:, 0, 0::2] = torch.sin(position * div_term)
        pe[:, 0, 1::2] = torch.cos(position * div_term)

        # 将 pe 注册为 buffer，这样它就不会被视为模型参数，但会随模型移动（例如 to(device)）
        self.register_buffer('pe', pe)

    def forward(self, x: torch.Tensor) -> torch.Tensor:
        # x.size(0) 是当前输入的序列长度
        # 将位置编码加到输入向量上
        x = x + self.pe[:x.size(0)]
        return self.dropout(x)
```

### 确认事项 / Verification

- [x] 我已阅读过相关章节的文档 / I have read the relevant chapter documentation
- [x] 我已搜索过现有的Issues,确认此问题未被报告 / I have searched existing Issues and confirmed this hasn't been reported
- [x] 我已尝试过基本的故障排除(如重启、重新安装依赖等) / I have tried basic troubleshooting (restart, reinstall dependencies, etc.)

**Comment by jjyaoao** [comment]
是不是章节写错了，应该是第三章对不对，你可以直接提一个PR来把你的想法给写在正文里，我会来review一下

**Comment by Sonnet1219** [comment]
> 是不是章节写错了，应该是第三章对不对，你可以直接提一个PR来把你的想法给写在正文里，我会来review一下

好的 是我章节写错了 不好意思

**Comment by jjyaoao** [comment]
感谢，已经merge
